rust_cli/src/commands/idl.rs

```diff
@@ -0,0 +1,512 @@
+    signer: &Keypair,
+    priority_fees_per_cu: u64,
+    seed: &str,
+) -> Result<()> {
+    let (_, rpc_client) = get_user_config()?;
+    
+    // Parse program ID
+    let program_pubkey = Pubkey::from_str(program_id)
+        .map_err(|e| anyhow!("Invalid program ID: {}", e))?;
+
+    // Derive PDA for IDL account
+    let (idl_address, _) = Pubkey::find_program_address(
+        &[seed.as_bytes(), program_pubkey.as_ref()],
+        &UPLOAD_IDL_ANCHOR_ID
+    );
+
+    // Check if account already exists
+    if let Ok(_) = rpc_client.get_account(&idl_address) {
+        println!("Data account already exists");
+        return Ok(());
+    }
+
+    println!("Initializing Data with seed: {}", seed);
+
+    // Get program data address
+    let (program_data_address, _) = Pubkey::find_program_address(
+        &[program_pubkey.as_ref()],
+        &solana_sdk::bpf_loader_upgradeable::ID,
+    );
+
+    // Create initialize instruction using the generated code
+    let accounts = Initialize {
+        idl: idl_address,
+        signer: signer.pubkey(),
+        system_program: solana_sdk::system_program::ID,
+        program_id: program_pubkey,
+        program_data: program_data_address,
+    };
+
+    let args = InitializeInstructionArgs {
+        seed: seed.to_string(),
+    };
+
+    let ix = accounts.instruction(args);
+
+    // Create and send transaction
+    let recent_blockhash = rpc_client
+        .get_latest_blockhash()
+        .map_err(|e| anyhow!("Failed to get recent blockhash: {}", e))?;
+
+
+    //TODO: find more elegant solution to set appropriate compute unit limit
+    let transaction_to_simulate = Transaction::new_signed_with_payer(
+        &[ComputeBudgetInstruction::set_compute_unit_price(priority_fees_per_cu), ComputeBudgetInstruction::set_compute_unit_limit(200000) ,ix.clone()],
+        Some(&signer.pubkey()),
+        &[signer],
+        recent_blockhash,
+    );
+
+    let simulation = rpc_client.simulate_transaction(&transaction_to_simulate)
+    .map_err(|e| anyhow!("Failed to simulate transaction: {}", e))?;
+
+    let units_consumed: u32 = simulation.value.units_consumed.unwrap_or(0) as u32;
+
+    println!("Units consumed: {}", units_consumed);
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[ComputeBudgetInstruction::set_compute_unit_price(priority_fees_per_cu), ComputeBudgetInstruction::set_compute_unit_limit(units_consumed + 200) ,ix],
+        Some(&signer.pubkey()),
+        &[&signer],
+        recent_blockhash,
+    );
+
+    // let mut config = RpcSendTransactionConfig::default();
+    // config.skip_preflight = true;
+
+    let signature = rpc_client
+        .send_and_confirm_transaction_with_spinner_and_commitment(&transaction, CommitmentConfig::confirmed())
+        .map_err(|e| anyhow!("Failed to send transaction: {}", e))?;
+
+    println!("Data initialized successfully!");
+    println!("Signature: {}", signature);
+    println!("Data Account: {}", idl_address);
```

Comment:
Maybe more speaking like: "Newly created metadata PDA address from seed {seed} : {address}" or smth like that